yb-voyager/cmd/live_migration_integration_test.go

```diff
@@ -150,142 +62,249 @@ FROM generate_series(%d, %d);`, startID, endId))
+	md5(random()::text) || '@example.com',                    -- email
+	repeat(md5(random()::text), 10)                           -- description (~320 chars)
+FROM generate_series(1, 5);`,
+		},
+		CleanupSQL: []string{
+			`DROP SCHEMA IF EXISTS test_schema CASCADE;`,
+		},
 	})
-	if err := postgresContainer.Start(ctx); err != nil {
-		utils.ErrExit("Failed to start Postgres container: %v", err)
-	}
 
-	// Start YugabyteDB container.
-	yugabytedbContainer := testcontainers.NewTestContainer("yugabytedb", nil)
-	if err := yugabytedbContainer.Start(ctx); err != nil {
-		utils.ErrExit("Failed to start YugabyteDB container: %v", err)
-	}
-	postgresContainer.ExecuteSqls([]string{
-		createSchemaSQL,
-		createTableSQL,
-		insertDataSQL,
-	}...)
-
-	yugabytedbContainer.ExecuteSqls([]string{
-		createSchemaSQL,
-		createTableSQL,
-	}...)
-
-	defer postgresContainer.ExecuteSqls(dropSchemaSQL)
-	defer yugabytedbContainer.ExecuteSqls(dropSchemaSQL)
-
-	err := testutils.NewVoyagerCommandRunner(postgresContainer, "export data", []string{
-		"--export-dir", exportDir,
-		"--source-db-schema", "test_schema",
-		"--disable-pb", "true",
-		"--export-type", SNAPSHOT_AND_CHANGES,
-		"--yes",
-	}, func() {
-		time.Sleep(5 * time.Second) // Wait for the export to start
-	}, true).Run()
-	testutils.FatalIfError(t, err, "Export command failed")
-
-	err = testutils.NewVoyagerCommandRunner(yugabytedbContainer, "import data", []string{
-		"--export-dir", exportDir,
-		"--disable-pb", "true",
-		"--yes",
-	}, func() {
-		time.Sleep(5 * time.Second)
-	}, true).Run()
-
-	testutils.FatalIfError(t, err, "Import command failed")
-
-	ok := utils.RetryWorkWithTimeout(1, 30, func() bool {
-		return snapshotPhaseCompleted(t, postgresContainer.GetConfig().Password,
-			yugabytedbContainer.GetConfig().Password, 10, `test_schema."test_live"`)
-	})
-	assert.True(t, ok)
+	defer lm.Cleanup()
 
-	// Connect to both Postgres and YugabyteDB.
-	pgConn, err := postgresContainer.GetConnection()
-	testutils.FatalIfError(t, err, "connecting to Postgres")
+	err := lm.SetupContainers(context.Background())
+	testutils.FatalIfError(t, err, "failed to setup containers")
 
-	ybConn, err := yugabytedbContainer.GetConnection()
-	testutils.FatalIfError(t, err, "Error connecting to YugabyteDB")
+	err = lm.SetupSchema()
+	testutils.FatalIfError(t, err, "failed to setup schema")
 
-	// Compare the full table data between Postgres and YugabyteDB for snapshot part.
-	// We assume the table "test_data" has a primary key "id" so we order by it.
-	if err := testutils.CompareTableData(ctx, pgConn, ybConn, "test_schema.test_live", "id"); err != nil {
-		t.Errorf("Table data mismatch between Postgres and YugabyteDB: %v", err)
-	}
+	err = lm.StartExportData(true, nil)
+	testutils.FatalIfError(t, err, "failed to start export data")
+
+	err = lm.StartImportData(true, nil)
+	testutils.FatalIfError(t, err, "failed to start import data")
+
+	lm.SnapshotTimeout = 30
```

Comment:
1. we can pass timeout in the `waitForSnapshotComplete` method itself. 
2. for the `waitForSnapshotComplete` method, relying on just one table is not going to work when we have multiple tables. Can we use some other deterministic metadb flag or something to determine when snapshot is complete?